app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,136 @@
+class OrdersController < ApplicationController
+  
+  before_action :authenticate_user!
+  before_action :set_journey, only: %i(new create checkout)
+  before_action :set_order, only: %i(checkout charge)
+  before_action :instantiate_order, only: %i(create)
+  before_action :ensure_booking_is_active, only: %i(new checkout create)
+  
+  def checkout
+  end
+  
+  def new
+    @minutes = calculate_minutes
+    @order = Order.new
+    authorize(@order)
+    # add here all the sum of all the minutes that have not been paid. 
+  end 
+  
+  def create
+    @order.consumer_total = @order.minutes * @order.price_hour / 60
```

Comment:
No need to notify me, once you update the code GH will mark/hide my comment as outdated